Improve the Azure cloud auto join documentation

pvandervelde opened the pull request:
Update the Azure cloud-auto-join documentation with more information on the configuration of the Azure resources that need to be applied to make the auto-join feature work. 

Linked to #8432.

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/consul?pullRequest=8530)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/consul?pullRequest=8530)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/consul?pullRequest=8530).</sub>

pvandervelde commented:
@jsosulska Here is the PR for the Azure cloud auto-join documentation improvement. 

I have a terraform example for setting up the Azure AD group and the role definitions etc. when using VMs. I don't think that suitable for this documentation but if it is a useful thing to have somewhere let me know which documents I should update (in a separate PR).

Also not sure about the exact wording, documentation isn't my strong suit. Let me know if it's not right and I'll attempt an improvement :)

jsosulska commented on website/pages/docs/agent/cloud-auto-join.mdx:
```diff
@@ -122,11 +122,11 @@ Use these configuration parameters (instead of `tag_name` and `tag_value`) when
 When using Virtual Machine Scale Sets the only role action needed is `Microsoft.Compute/virtualMachineScaleSets/*/read`.
 
-~> **Note:** If the Consul datacenter is hosted on Azure, Consul can use Managed Service Identities (MSI) to access Azure instead of an environment variable and shared client id and secret. MSI must be enabled on the VMs hosting Consul, and it is the preferred configuration since MSI prevents your Azure credentials from being stored in Consul configuration. This feature is supported from Consul 1 ...
+~> **Note:** If the Consul datacenter is hosted on Azure, Consul can use Managed Service Identities (MSI) to access Azure instead of an environment variable and shared client id and secret. MSI must be enabled on the VMs or Virtual Machine Scale Sets hosting Consul, and it is the preferred configuration since MSI prevents your Azure credentials from being stored in Consul configuration. This feat ...
```
```suggestion
~> **Note:** If the Consul datacenter is hosted on Azure, Consul can use Managed Service Identities (MSI) to access Azure instead of an environment variable, shared client id and secret. MSI must be enabled on the VMs or Virtual Machine Scale Sets hosting Consul, It is the preferred configuration since MSI prevents your Azure credentials from being stored in Consul configuration. This feature is supported in Consul 1.7 and above. When using MSI, the `tag_key`, `tag_value` and `subscription_id` need to be supplied for Virtual machines.
```

Great job here. I cleaned up the wording from before, and changed `tag_name` to match `tag_key` above.

pvandervelde replied:
Thanks. I've taken your suggestion. The only thing I changed is a comma to period here

    MSI must be enabled on the VMs or Virtual Machine Scale Sets hosting Consul, It is the

I changed it to

    MSI must be enabled on the VMs or Virtual Machine Scale Sets hosting Consul. It is the

Is that the correct way around or did you mean to lower case the `It`?

pvandervelde commented:
@jsosulska My gist for the terraform files are here: https://gist.github.com/pvandervelde/0c9a805e83fd4bc6e51fefbf6ee52c51. These are a copy + simplification of my code in the repository / branch / file here: https://github.com/Calvinverse/infrastructure.azure.core.servicediscovery/blob/feature/initial/src/main.tf. The code in my repository works, but I might have introduced bugs in the gist as I simplified with out testing (oops)

pvandervelde commented:
I've also just noticed that when using MSI's it may initially take a while before the permissions are synced / available. I just created a consul server cluster with terraform and it took about an hour before Azure noticed that my MSIs did have the right permissions. Not super impressive time-wise. I have however had it work really quickly as well.
Should I add to the note that it may take time?

pvandervelde commented:
@jsosulska I've updated the documentation page with your suggestion and slightly tweaked it to indicate that it is the VMs that don't seem to pick up the permissions. I've also split the original sentence over multiple lines in the markdown because it was getting very long. Let me know if that breaks things or is not according to your standards.

jsosulska reviewed (changes requested):
Looks great so far! I noticed one small change - 

On L#92 and L#98 - can you change `tag_name` to `tag_key` to match the description?

Other than that change, LGTM!

pvandervelde commented:
@jsosulska I've made the two changes you requested. My apologies for taking so long.

jsosulska commented:
LGTM!

jsosulska reviewed (approved):
LGTM
